server/src/main/java/org/apache/druid/segment/realtime/appenderator/SinkQuerySegmentWalker.java

```diff
@@ -92,6 +94,8 @@ public class SinkQuerySegmentWalker implements QuerySegmentWalker
   private final Cache cache;
   private final CacheConfig cacheConfig;
   private final CachePopulatorStats cachePopulatorStats;
+  private final ConcurrentMap<SegmentDescriptor, SegmentDescriptor> newIdToBasePendingSegment
```

Comment:
I don't see items ever being removed from this map. But I guess that will never be needed?

Reply:
Yes. We could do it when a segment is unannounced but it will not be queried subsequently and it isn't likely for there to be a significant overhead due to such entries